src/Chainweb/Mempool/InMem.hs

```diff
@@ -241,6 +255,12 @@ insertInMem cfg lock runCheck txs0 = do
     sizeOK tx = getSize tx <= maxSize
     maxRecent = _inmemMaxRecentItems cfg
     hasher = txHasher txcfg
+    ttlCheck tx =
+      case txMetadata txcfg tx of
+        TransactionMetadata (Time (TimeSpan creationTime)) (Time (TimeSpan expiryTime)) ->
+            creationTime >= 0 && expiryTime >= 0 && creationTime < expiryTime
+
+    preGossipCheck' = txPreGossipCheck txcfg
```

Comment:
I think we can delete this now that we have a pre-insert check in InMem?

Reply:
Done.